### ISSUE TITLE
Audit Calico Addon

### ISSUE THREAD

**Issue by kcoleman731** [shapirov103/cdk-eks-blueprint#49]
@shapirov103 lets talk about this addon. First lets get rid of the hard coded yaml file in favor of this helm chart 

https://artifacthub.io/packages/helm/aws/aws-calico

Also, lets talk about what sensible defaults we can provide.

**Comment by shapirov103** [comment]
The question is: should not we use the official AWS documentation on Calico installation or do we favor a helm chart? The overhead from the maintenance perspective is the same. Plus the helm chart is more around installing Calico in full CNI mode, while the only pertinent distinction is `helm upgrade --install --recreate-pods --force aws-calico --namespace kube-system eks/aws-calico`. They also have a dependency on CRDs being installed for which they point to github repository which i don't I can keep.

**Comment by parkand1** [comment]
I personally think we should stick to the official AWS documentation on how to install Calico to provide a consistent experience for customers. I think it would provide the best experience for customers the less they have to worry about dependencies and the more they can rely on what we will end up providing out of the box.

**Comment by shapirov103** [comment]
Right, I was leaning towards the AWS documented way since I am sure it is tested (hopefully continuously). The helm chart is probably also tested, but they don't provide a version matrix.

**Comment by kcoleman731** [comment]
I'm dumb. Looks like we have official calico charts https://github.com/aws/eks-charts/tree/master/stable/aws-calico
